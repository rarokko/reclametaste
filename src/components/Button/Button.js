import React from 'react';
import './Button.scss';

export function Button(props) {

  return (
    <button onClick={props.onClick} className="rt-button">
      {props.children}
    </button>
  );
}