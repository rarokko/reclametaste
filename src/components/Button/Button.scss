@import '../../helpers/CommonStylesheet/Common';

.rt-button {
    width: 100%;
    background-color: #fff;
    border: 1px solid #bfbfbf;
    padding: 13px;
    min-height: 47px;
    border-radius: 5px;
    font-size: 1em;
    font-weight: normal;
    text-transform: uppercase;
    color: $color-salmon;
    box-sizing: border-box;
    cursor: pointer;
}