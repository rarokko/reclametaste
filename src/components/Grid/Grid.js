import React from 'react';
import './Grid.scss';

export function Grid(props) {

  return (
    <div
      className="rt-grid-container"
    >
      {props.children.map((item, index) => {
        return(
          <div className="rt-grid-item" key={`item_${index}`}>
            {item}
          </div>
        )
      })}
    </div>
  );
}