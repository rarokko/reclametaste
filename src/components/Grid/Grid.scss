@import '../../helpers/CommonStylesheet/Common';

.rt-grid-container {
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(250px, 1fr));
    grid-gap: 30px;
    width: 100%;
}

.rt-grid-item {
    background-color: #f9f9f9;
    border-radius: 5px;
    position: relative;
    figure {
        margin: 0;
    }
    img {
        width: 100%;
        display: block;
        height: 250px;
        border-radius: 10px;
        object-fit: cover;
    }
    figcaption {
        position: absolute;
        bottom: -10px;
        left: 0;
        right: 0;
        margin: 0 auto;
        padding: 15px;
        border-radius: 10px;
        font-size: 1em;
        background-color: #ffffff;
        width: 80%;
        color: $color-salmon;
    }
    i {
        position: absolute;
        top: 0;
        left: 0;
        color: $color-salmon;
        font-size: 2em;
        background-color: #fff;
        border-radius: 0px 0px 15px 0px;
        width: 1.5em;
        height: 1.5em;
        display: flex;
        align-items: center;
        justify-content: center;
        cursor: pointer;
    }
}