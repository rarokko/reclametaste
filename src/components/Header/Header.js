import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import Request from '../../helpers/Request/Request';
import './Header.scss';

export class Header extends Component {

  constructor() {
    super();

    let pathname = window.location.pathname;
    pathname = pathname.indexOf('/', 1) > 0 ? pathname.substring(1, pathname.indexOf('/', 1)) : pathname.substring(1, pathname.length);

    this.state = {
      activePage: pathname !== "" ? pathname : 'marvel'
    };
  };

  logout() {
    Request.logoutUser();
  }

  handlePageChange(pathname) {

    this.setState((state, props) => {
      return {
        activePage: pathname
      }
    });
  }

  render() {
    return (
      <header>
        <nav className="rt-header-nav">
          <div>
            <Link to="/marvel" onClick={() => this.handlePageChange('marvel')} >
              <button data-active={this.state.activePage === "marvel" ? 'true' : 'false'}>Marvel</button>
            </Link>
            <Link to="/beer" onClick={() => this.handlePageChange('beer')} >
              <button data-active={this.state.activePage === "beer" ? 'true' : 'false'}>Beer</button>
            </Link>
          </div>
          <button onClick={() => this.logout()}>Logout</button>
        </nav>
      </header>
    )
  }
}