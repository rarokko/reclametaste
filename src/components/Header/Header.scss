@import '../../helpers/CommonStylesheet/Common';

.rt-header-nav {
    display: flex;
    background-color: #fff;
    flex-direction: column;
    align-items: center;
    position: relative;
    
    @media #{$min-big-screen} {
        flex-direction: row;
        justify-content: space-between;
    }

    &>:last-child {
        font-size: 1em;
        padding: 7px;
        position: absolute; 
        bottom: -42px;
        border-radius: 0px 0px 10px 10px;
        z-index: 5;
        
        @media #{$min-big-screen} {
            order: 0;
            font-size: 1.5em;
            bottom: -52px;
            right: 0;
        }
    }

    button {
        border: none;
        color: $color-salmon;
        font-size: 1.5em;
        text-transform: uppercase;
        font-weight: bold;
        padding: 15px;
        margin: 10px;
        background-color: #fff;
        border-radius: 10px;
        cursor: pointer;
    }
    
    button[data-active=true] {
        background-color: $color-salmon;
        color: #ffffff;
    }

    button:hover {
        background-color: #e8e8e8;
    }
}