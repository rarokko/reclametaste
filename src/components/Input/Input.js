import React from 'react';
import './Input.scss';

export function Input(props) {

  return (
      <input 
        value={props.value} 
        onChange={props.onChange}
        className="rt-input" 
        type={props.type} 
        placeholder={props.placeholder}
        required={props.required} 
      />
  );
}