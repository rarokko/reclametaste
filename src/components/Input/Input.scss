@import '../../helpers/CommonStylesheet/Common';

.rt-input {
    width: 100%;
    padding: 15px;
    border-radius: 5px;
    border: 1px solid #bfbfbf;
    color: #636363;
    margin-bottom: 10px;
    box-sizing: border-box;
    display: inline-block;
    vertical-align: top;
}