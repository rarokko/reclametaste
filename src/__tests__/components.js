import React from 'react';
import {HomeUnlogged} from '../containers/HomeUnlogged/HomeUnlogged';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Input } from '../components/Input/Input';
import { Button } from '../components/Button/Button';
import { Select } from '../components/Select/Select';
import { Grid } from '../components/Grid/Grid';
import { Loader } from '../components/Loader/Loader';
import { Header } from '../components/Header/Header';

describe('Button', () => {
  it('render properly', () => {
    const component = shallow(<Button/>);
    expect(component).toMatchSnapshot();
  });
});

describe('Grid', () => {
  it('render properly', () => {
    const component = shallow(
    <Grid>
      <div/>
      <div/>
      <div/>
    </Grid>
      );
    expect(component).toMatchSnapshot();
  });
});

describe('Header', () => {
  it('render properly', () => {
    const component = shallow(<Header/>);
    expect(component).toMatchSnapshot();
  });
});

describe('Input', () => {
  it('render properly', () => {
    const component = shallow(<Input/>);
    expect(component).toMatchSnapshot();
  });
});

describe('Loader', () => {
  it('render properly', () => {
    const component = shallow(<Loader/>);
    expect(component).toMatchSnapshot();
  });
});

describe('Select', () => {
  it('render properly', () => {
    const component = shallow(<Select/>);
    expect(component).toMatchSnapshot();
  });
});