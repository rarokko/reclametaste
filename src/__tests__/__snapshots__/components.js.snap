// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render properly 1`] = `
<button
  className="rt-button"
/>
`;

exports[`Grid render properly 1`] = `
<div
  className="rt-grid-container"
>
  <div
    className="rt-grid-item"
    key="item_0"
  >
    <div />
  </div>
  <div
    className="rt-grid-item"
    key="item_1"
  >
    <div />
  </div>
  <div
    className="rt-grid-item"
    key="item_2"
  >
    <div />
  </div>
</div>
`;

exports[`Header render properly 1`] = `
<header>
  <nav
    className="rt-header-nav"
  >
    <div>
      <Link
        onClick={[Function]}
        to="/marvel"
      >
        <button
          data-active="true"
        >
          Marvel
        </button>
      </Link>
      <Link
        onClick={[Function]}
        to="/beer"
      >
        <button
          data-active="false"
        >
          Beer
        </button>
      </Link>
    </div>
    <button
      onClick={[Function]}
    >
      Logout
    </button>
  </nav>
</header>
`;

exports[`Input render properly 1`] = `
<input
  className="rt-input"
/>
`;

exports[`Loader render properly 1`] = `
<div
  className="rt-loader"
/>
`;

exports[`Select render properly 1`] = `
<select
  className="rt-select"
/>
`;
